packages/opentelemetry-resources/src/types.ts

```diff
@@ -26,9 +26,10 @@ import { SpanAttributes } from '@opentelemetry/api';
 export type ResourceAttributes = SpanAttributes;
 
 /**
- * Interface for a Resource Detector. In order to detect resources in parallel
- * a detector returns a Promise containing a Resource.
+ * Interface for a Resource Detector. In order to detect resources asynchronously, a detector
+ * can pass a Promise as the second parameter to the Resource constructor. Returning a
+ * Promise<Resource> is deprecated in favor of this approach.
  */
 export interface Detector {
-  detect(config?: ResourceDetectionConfig): Promise<Resource>;
+  detect(config?: ResourceDetectionConfig): Promise<Resource> | Resource;
```

Comment:
We talked about it in the SIG. Since it's a breaking change for the public interface on stable package, we should create a new interface for which the detect function returns `Resource`

Reply:
Should I update the detectors that are already sync (eg envDetector) to use the new sync interface?